src/simulation.jl

```diff
@@ -143,45 +165,85 @@ end
 
 is_form(d::SummationDecapode, var_name::Symbol) = is_form(d, first(incident(d, var_name, :name)))
 
+function getgeneric_type(type::Symbol)
+  if (type == :Form0 || type == :Form1 || type == :Form2 ||
+    type == :DualForm0 || type == :DualForm1 || type == :DualForm2)
+    return :Form
+  end
+  return type
+end
+
 is_literal(d::SummationDecapode, var_id::Int) = (d[var_id, :type] == :Literal)
 is_literal(d::SummationDecapode, var_name::Symbol) = is_literal(d, first(incident(d, var_name, :name)))
 
 is_infer(d::SummationDecapode, var_id::Int) = (d[var_id, :type] == :infer)
 is_infer(d::SummationDecapode, var_name::Symbol) = is_infer(d, first(incident(d, var_name, :name)))
 
-add_stub(stub_name::Symbol, var_name::Symbol) = return Symbol("$(stub_name)_$(var_name)")
+struct InvalidStubException <: Exception
+  stub::Symbol
+end
+
+Base.showerror(io::IO, e::InvalidStubException) = print(io, "Stub \"$(e.stub)\" is invalid")
+
+function add_stub(stub_name::Symbol, var_name::Symbol)
+  # No empty stubs
+  if stub_name == Symbol() || stub_name == NO_STUB_RETURN || !isascii(String(stub_name))
+    throw(InvalidStubException(stub_name))
+  end
+
+  return Symbol("$(stub_name)_$(var_name)")
+end
 
+# ! Warning: Bad behavior if not stub was added and  variable has an "_" itself
 function get_stub(var_name::Symbol)
   var_str = String(var_name)
   idx = findfirst("_", var_str)
-  if(isnothing(idx) || first(idx) == 1)
-    return nothing
+
+  if(isnothing(idx))
+    return NO_STUB_RETURN
+  end
+
+  if(first(idx) == 1)
+    throw(InvalidStubException(var_name))
   end
+
   return Symbol(var_str[begin:first(idx) - 1])
 end
 
-add_inplace_stub(var_name::Symbol) = add_stub(gensim_in_place_stub, var_name)
+add_inplace_stub(var_name::Symbol) = add_stub(GENSIM_INPLACE_STUB, var_name)
+
+const ARITHMETIC_OPS = Set([:+, :*, :-, :/, :.+, :.*, :.-, :./, :^, :.^, :.>, :.<, :.≤, :.≥])
+
+struct InvalidCodeTargetException <: Exception
+  code_target::GenerationTarget
+end
+
+Base.showerror(io::IO, e::InvalidCodeTargetException) = print(io, "Provided code target $(e.code_target) is not yet supported in simulations")
 
-# This will be the function and matrix generation
-function compile_env(d::AbstractNamedDecapode, dec_matrices::Vector{Symbol}, con_dec_operators::Set{Symbol}, code_target::GenerationTarget = CPUTarget())
-  assumed_ops = Set([:+, :*, :-, :/, :.+, :.*, :.-, :./, :^, :.^, :.>, :.<, :.≤, :.≥])
-  defined_ops = Set()
+"""
+    compile_env(d::SummationDecapode, dec_matrices::Vector{Symbol}, con_dec_operators::Set{Symbol}, code_target::GenerationTarget = CPUTarget())
```

Comment:
This docstring ways that `code_target` defaults to `CPUTarget`, but the actual function signature does not appear to do that

Reply:
Fixed.